dask/async.py

```diff
@@ -414,18 +421,16 @@ def get_async(apply_async, num_workers, dsk, result, cache=None,
 
     state = start_state_from_dask(dsk, cache=cache)
 
-    tick = [0]
-
     if state['waiting'] and not state['ready']:
         raise ValueError("Found no accessible jobs in dask")
 
+    if not execute_cm:
+        execute_cm = default_cm
+
     def fire_task():
         """ Fire off a task to the thread pool """
-        # Update heartbeat
-        tick[0] += 1
-        # Emit visualization if called for
-        if debug_counts and tick[0] % debug_counts == 0:
-            visualize(dsk, state, filename='dask_%03d' % tick[0])
+        if scheduler_callback:
+            scheduler_callback(dsk, state)
```

Comment:
I can envision applications wanting to know the key that is being executed when this is called.